spec/bolt/result_spec.rb

```diff
@@ -8,6 +8,33 @@
 describe Bolt::Result do
   let(:target) { "foo" }
 
+  describe :initialize do
+    it 'sets default values' do
+      result = Bolt::Result.new(target)
+      expect(result.target).to eq('foo')
+      expect(result.value).to eq({})
+      expect(result.action).to eq('action')
+      expect(result.object).to eq(nil)
+    end
+
+    it 'sets error' do
+      result = Bolt::Result.new(target, error: { 'This' => 'is an error' })
+      expect(result.error_hash).to eq('This' => 'is an error')
+      expect(result.value['_error']).to eq('This' => 'is an error')
+    end
+
+    it 'errors if error is not a hash' do
+      expect { Bolt::Result.new(target, error: 'This is an error') }
+        .to raise_error(RuntimeError, 'TODO: how did we get a string error')
```

Comment:
Oh my gosh ha thanks for adding this! Finally testing it after 2 years https://github.com/puppetlabs/bolt/blame/44fbef4d2279298167745b3e3d92be323b9769e1/lib/bolt/result.rb#L99